lib/smack/SmackInstGenerator.cpp

```diff
@@ -623,6 +623,16 @@ void SmackInstGenerator::visitCallInst(llvm::CallInst& ci) {
     Value* val = ci.getArgOperand(0);
     emit(Stmt::assign(rep->expr(&ci), rep->expr(val)));
 
+  } else if (SmackOptions::BitPrecise && SmackOptions::FloatEnabled &&
```

Comment:
I would like that all intrinsics are then handled using models from a C header. So you should refactor `llvm.expect` above to to the same. Like I said, either we should model all intrinsics in a C header, or we should model all intrinsics directly in C++ (meaning here). Now are are doing both, which I don't like since it is not uniform.

Reply:
I agree. However, since the scope of this PR is limited to half-conversion intrinsics, maybe we want another PR to implement a file that handles all the LLVM intrinsics?

Reply:
We'd still be kind of inconsistent with intrinsics since we're writing some of them directly in boogie, while the integer overflow intrinsics are written as LLVM transformations. I suppose this is okay since most overflow checks are handled fairly differently, except for the division and left shift overflow checks, which sort of amount to what we're doing here. We also have the special functions (rust panic, etc.) which we probably want to organize as well.

Reply:
Ok, I am fine with another pull request that will refactor this code as I suggested.
Can you do that right after this one?

Reply:
Yes. That way we can see what the new structure looks like.